### PR TITLE
Update Julia to Julia 1.0

### PR DESCRIPTION
**Pull request by odow**
image

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
+LinQuadOptInterface = "f8899e07-120b-5979-ab1d-7b97bb9e1a48"
+LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+MathProgBase = "fdba3010-5040-5b88-9595-932c9decdf73"
+SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
+
+[compat]
+LinQuadOptInterface = "~0.6.0"
+MathProgBase = "~0.5.0, ~0.6, ~0.7"
+julia = "≥ 1"
```
Comment (mlubin) [inline Project.toml]: "1" so that we don't claim to support Julia 2.0.

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name = "Gurobi"
+uuid = "2e9cd046-0924-5485-92f1-d5272153d98b"
```
Comment (mlubin) [inline Project.toml]: Missing repo and version.

File: src/MPB_wrapper.jl [hunk 3]
```diff
@@ -116,7 +116,7 @@ function MPB.loadproblem!(m::GurobiMathProgModel, filename::AbstractString)
 end
 
 if VERSION >= v"0.7-"
```
Comment (mlubin) [inline src/MPB_wrapper.jl]: This version check can be removed.

File: src/grb_constrs.jl [hunk 4]
```diff
@@ -79,7 +75,7 @@ end
 function add_constrs_t!(model::Model, At::Matrix{Float64}, rel::GCharOrVec, b::Vector)
     n, m = size(At)
     (m == length(b) && n == num_vars(model)) || error("Incompatible argument dimensions.")
-    add_constrs_t!(model, sparse(At), rel, b)
+    add_constrs_t!(model, SparseArrays.sparse(At), rel, b)
```
Comment (mlubin) [inline src/grb_constrs.jl]: `sparse` is used frequently enough that we may want `using SparseArrays`.

Comment (odow) [reply src/grb_constrs.jl]: I've left for now.

**Comment by odow** [comment]
Once this is merged, I'm going to have a go at removing LQOI and writing a direct MOI v9 wrapper so see how much we can strip out.

File: Project.toml [hunk 5]
```diff
@@ -11,7 +13,7 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 [compat]
 LinQuadOptInterface = "~0.6.0"
 MathProgBase = "~0.5.0, ~0.6, ~0.7"
-julia = "≥ 1"
+julia = "~1"
```
Comment (mlubin) [inline Project.toml]: I think "1" is sufficient to include all 1.x releases.

Comment (odow) [reply Project.toml]: I haven't quite got my head around this new format.

**Comment by blegat** [comment]
> Once this is merged, I'm going to have a go at removing LQOI and writing a direct MOI v9 wrapper so see how much we can strip out.

What are the rationale for this ? The plan is to stop using LQOI from Gurobi or remove the need for LQOI everywhere ? I am not an expert of LQOI so I cannot judge what's the best solution. For SDOI, variable bridge will considerably simplify it so it may be possible to get rid of it after that.

**Comment by odow** [comment]
The plan is to see how much more readable the Gurobi wrapper is without LQOI (one less layer). Now that MOI is more stable, a little duplication is probably better than an extra layer/dependency.

**Comment by mlubin** [comment]
My impression is that many of the cases that motivated LQOI are now covered by bridges.

**Pull request merged**